### PR TITLE
Add service details

### PR DESCRIPTION
**Pull request by TimothyAsirJeyasing**
Detect the service details and populate that into etcd.

tendrl-bug-id:Tendrl/specifications#73
Signed-off-by: Timothy Asir J <tjeyasin@redhat.com>

### CODE REVIEW

**Comment by TimothyAsir** [comment]
@r0h4n , @nthomas-redhat , @shtripat , Please review

File: tendrl/node_agent/manager/pull_hardware_inventory.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 from command import Command
 import platform
 import socket
+from service_deduction import Service
```
Comment (shtripat) [inline tendrl/node_agent/manager/pull_hardware_inventory.py]: deduction??

File: tendrl/node_agent/manager/service_detection.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
+from command import Command
+import logging
+import psutil
+from tendrl.node_agent.config import TendrlConfig
+from tendrl.node_agent.manager import utils as mgr_utils
+
+config = TendrlConfig()
+LOG = logging.getLogger(__name__)
+
+NAMES=["httpd", "glusterd", "sshd", "tendrl-node-agent"]
```
Comment (shtripat) [inline tendrl/node_agent/manager/service_detection.py]: %s/NAMES/VALID_SERVICES/g

File: tendrl/node_agent/manager/service_detection.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        # This can be used to set the names externally
+        self.names = names
+
+    def get_details(self):
+        '''returns structure
+        {"name":           "Service Name",
+        "command":        "Execution Script",
+        "exe_path":       "Execution Path",
+        "status":         "Service Status",
+        "cpu_usage_per":  "Cpu Usage Percentage",
```
Comment (shtripat) [inline tendrl/node_agent/manager/service_detection.py]: %s/cpu_usage_per/cpu_usage_percent/g

File: tendrl/node_agent/manager/service_detection.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        "threads_count":  "Number of Threads",
+        "open_files_cnt": "Number of open files"}
+        '''
+
+        services = []
+        rv = {}
+        try:
+            pDet = [psutil.Process(p) for p in psutil.get_pid_list() \
+                    if psutil.Process(p).name() in self.names]
+        except psutil.NoSuchProcess as e:
```
Comment (shtripat) [inline tendrl/node_agent/manager/service_detection.py]: Log at least

**Comment by r0h4n** [comment]
@TimothyAsir let me know if this is to be re-opened

**Pull request closed without merging**